### PR TITLE
add exception handling to ollama client inference

### PR DESCRIPTION
**Pull request by varshney-yash**

### CODE REVIEW

**Review by KenAijmNight — COMMENTED** [review]
Where are the Docs?

File: src/llm/ollama_client.py [hunk 1]
```diff
@@ -3,23 +3,24 @@
 
 from src.logger import Logger
 
+logger = Logger()
```
Comment (mufeedvh) [inline src/llm/ollama_client.py:6]: Initiate the logger inside the class `__init__`?

**Review by mufeedvh — CHANGES_REQUESTED** [review]

File: src/llm/ollama_client.py (hunk 1 above)
Comment (varshney-yash) [reply src/llm/ollama_client.py:6]: list_models is a staticmethod, can't pass self in it

Comment (jerome-white) [reply src/llm/ollama_client.py:7]: Based on `list_models` only usage [here](https://github.com/stitionai/devika/blob/523cbcf709b21a385c91d928a37b20568a8c38f6/src/llm/llm.py#L30), it should also work as a classmethod. In that case, these global variables could be moved into the class (as static members).

**Comment by mufeedvh** [comment]
LGTM, merging this now. Thanks! ❤️ 🙌

**Pull request merged**